src/ert/ensemble_evaluator/evaluator.py

```diff
@@ -144,6 +145,8 @@ def _cancelled_handler(self, events: List[CloudEvent]) -> None:
             )
             send_future.result()
             self._loop.call_soon_threadsafe(self._stop)
+            while not self._done.done():
+                time.sleep(0.1)
```

Comment:
As this is repeated three times; should we put it in a separate method called `wait_until_done` or `sleep_until_done`?

Reply:
Yes, we should.

Reply:
I have replaced the `time.sleep` with `threading.Event()`